client-v2/src/main/java/com/clickhouse/client/api/data_formats/RowBinaryWithNamesAndTypesFormatReader.java

```diff
@@ -28,18 +28,15 @@ public RowBinaryWithNamesAndTypesFormatReader(InputStream inputStream, QuerySett
 
     private void readSchema() {
         try {
-            if (inputStream.available() < 1) {
-                return;
-            }
             TableSchema headerSchema = new TableSchema();
             List<String> columns = new ArrayList<>();
-            int nCol = chInputStream.readVarInt();
+            int nCol = BinaryStreamReader.readVarInt(input);

```

Comment:
- chInputStream.readVarInt() was here because com.clickhouse.data.ClickHouseInputStream#readVarInt is better implementation of com.clickhouse.data.format.BinaryStreamUtils#readVarInt(java.io.InputStream) (last one assumes only 9 byte while today server sends 10 bytes and input stream implementation works correct. 
- BinaryStreamUtils works only with ClickHouse streams what are the main drawback. I just reimplemented to be able to read from any stream (or buffer in the future)
-